src/join.t.sol

```diff
@@ -553,6 +579,57 @@ contract DssDeployTest is DssDeployTestBase {
         assertEq(exitbal, 99999800);
     }
 
+    function testFailGemJoin8JoinWad() public {
+        deployKeepAuth();
+        DSValue pip = new DSValue();
+        GUSD gusd = new GUSD(100 * 10 ** 2);
+        GemJoin8 gusdJoin = new GemJoin8(address(vat), "GUSD", address(gusd));
+        dssDeploy.deployCollateral("GUSD", address(gusdJoin), address(pip));
+        gusd.approve(address(gusdJoin), uint(-1));
+        // Fail here
+        gusdJoin.join(address(this), 10 ether);
+    }
+
+    function testFailGemJoin8ExitWad() public {
+        deployKeepAuth();
+        DSValue pip = new DSValue();
+        GUSD gusd = new GUSD(100 * 10 ** 2);
+        GemJoin8 gusdJoin = new GemJoin8(address(vat), "GUSD", address(gusd));
+        dssDeploy.deployCollateral("GUSD", address(gusdJoin), address(pip));
+        gusd.approve(address(gusdJoin), uint(-1));
+        gusdJoin.join(address(this), 10 * 10 ** 2);
+        // Fail here
+        gusdJoin.exit(address(this), 10 ether);
+    }
+
+    function testFailGemJoin8Join() public {
+        deployKeepAuth();
+        DSValue pip = new DSValue();
+        GUSD gusd = new GUSD(100 * 10 ** 2);
+        GemJoin8 gusdJoin = new GemJoin8(address(vat), "GUSD", address(gusd));
+        dssDeploy.deployCollateral("GUSD", address(gusdJoin), address(pip));
+        gusd.approve(address(gusdJoin), uint(-1));
+        assertEq(gusd.balanceOf(address(this)), 100 * 10 ** 2);
+        assertEq(gusd.balanceOf(address(gusdJoin)), 0);
+        assertEq(vat.gem("GUSD", address(this)), 0);
+        gusd.setImplementation(address(1));
+        // Fail here
```

Comment:
I think these `Fail` tests are failing on the previous line.

The function signature of the setter is `gusd.setImplementation(address, uint256);`, so these tests would fail on the implementation set, rather than the join.

Reply:
Ah, nevermind, the test GUSD is using a simplified setter. Disregard.

Reply:
This is setting the implementation on the `gem` instead of the `gemJoin`. They just happen to have the same method name. It should be correct behaviour.

Reply:
Just double checked and commenting out the failure line results in the test passing, so it is correct.